tests/pytests/test_vecsim.py

```diff
@@ -1383,23 +1383,129 @@ def test_timeout_reached():
+    conn = getConnectionByEnv(env)
+    dim = 4
+    n = 100
+    per_doc = 5
+    info_type = int if env.isCluster() else str
+
+    env.expect('FT.CREATE', 'idx', 'ON', 'JSON', 'SCHEMA',
+               '$.vecs[*]', 'AS', 'hnsw', 'VECTOR', 'HNSW', '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2',
+               '$.vecs[*]', 'AS', 'flat', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2').ok()
+
+    for i in range(n):
+        conn.json().set(i, '.', {'vecs': [[0.46 for _ in range(dim)] for _ in range(per_doc)]})
+
+    for _ in env.retry_with_rdb_reload():
+        info = conn.ft('idx').info()
+        env.assertEqual(info['num_docs'], info_type(n))
+        env.assertEqual(info['num_records'], info_type(n * per_doc * len(info['attributes'])))
+        env.assertEqual(info['hash_indexing_failures'], info_type(0))
+
+        cmd = ['FT.SEARCH', 'idx', '', 'PARAMS', '2', 'b', '????' * dim, 'NOCONTENT']
+
+        cmd[2] = '*=>[KNN 10 @hnsw $b]'
+        hnsw_res = conn.execute_command(*cmd)[1:]
+        env.assertEqual(len(hnsw_res), len(np.unique(hnsw_res)))
+
+        cmd[2] = '*=>[KNN 10 @flat $b]'
+        flat_res = conn.execute_command(*cmd)[1:]
+        env.assertEqual(len(flat_res), len(np.unique(flat_res)))
+
+def test_bad_index_multi_value_json():
+    env = Env(moduleArgs='DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+    info_type = int if env.isCluster() else str
+    dim = 4
+    per_doc = 5
+
+    failures = 0
+
+    env.expect('FT.CREATE', 'idx', 'ON', 'JSON', 'SCHEMA',
+               '$.vecs', 'AS', 'vecs', 'VECTOR', 'HNSW', '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2').ok()
+
+    # By default, we assume that a static path leads to a single value, so we can't
+    conn.json().set(46, '.', {'vecs': [[0.46 for _ in range(dim)] for _ in range(per_doc)]})
+    failures += 1
+    env.assertEqual(conn.ft('idx').info()['hash_indexing_failures'], info_type(failures))
+
+    # We also don't support an array of length 1 that wraps an array for single value
+    conn.json().set(46, '.', {'vecs': [[0.46 for _ in range(dim)]]})
+    failures += 1
+    env.assertEqual(conn.ft('idx').info()['hash_indexing_failures'], info_type(failures))
+
+    conn.flushall()
+    failures = 0
+    env.expect('FT.CREATE', 'idx', 'ON', 'JSON', 'SCHEMA',
+               '$.vecs[*]', 'AS', 'vecs', 'VECTOR', 'HNSW', '6', 'TYPE', 'FLOAT32', 'DIM', dim, 'DISTANCE_METRIC', 'L2').ok()
+
+    # dynamic path returns a non array type
+    conn.json().set(46, '.', {'vecs': [np.ones(dim).tolist(), 'not a vector']})
+    failures += 1
+    env.assertEqual(conn.ft('idx').info()['hash_indexing_failures'], info_type(failures))
+
+    # we should NOT fail if some of the vectors are NULLs
+    conn.json().set(46, '.', {'vecs': [np.ones(dim).tolist(), None]})
+    env.assertEqual(conn.ft('idx').info()['hash_indexing_failures'], info_type(failures))
+    env.assertEqual(conn.ft('idx').info()['num_records'], '1')
```

Comment:
In cluster, `ft.info` returns `1` as int apparently...

Reply:
Missed this one. Should be passed to `info_type`